BHoM_UI/Components/oM/CreateCustom.cs

```diff
@@ -79,9 +79,17 @@ public CreateCustomCaller() : base()
         /**** Public Methods              ****/
         /*************************************/
 
-        public void SetInputs(List<string> names)
+        public void SetInputs(List<string> names, List<Type> types = null)
         {
-            InputParams = names.Select(x => GetParam(x)).ToList();
+            if (names.Count != types.Count)
+            {
+                Engine.Reflection.Compute.RecordWarning("The list length for names and types does not match. Inputs are set, but <types> variable will be ignored.");
+                InputParams = names.Select(x => GetParam(x)).ToList();
+            }
+            else
+            {
+                InputParams = names.Select((x, i) => GetParam(names[i], types[i])).ToList();
```

Comment:
Not the biggest deal but there is a specific Linq method for that:
`InputParams = names.Zip(types, (name, type) => GetParam(name, type)).ToList();`

Reply:
Nice! Thanks, I didn't know that!